codersdk/idpsync.go

```diff
@@ -12,6 +12,15 @@ import (
 	"golang.org/x/xerrors"
 )
 
+// constraining to `uuid.UUID | string` here would be nice but `make gen` will
+// yell at you for it.
+type IDPSyncMapping[ResourceIdType any] struct {
```

Comment:
Fixed! 

https://github.com/coder/guts/commit/cffad1be9721b82b1dacde529ed5af316e184ce1

I'll push an update to this branch :+1: